Actions/Deploy/Deploy.ps1

```diff
@@ -5,32 +5,56 @@ Param(
     [string] $token,
     [Parameter(HelpMessage = "Specifies the parent telemetry scope for the telemetry signal", Mandatory = $false)]
     [string] $parentTelemetryScopeJson = '7b7d',
-    [Parameter(HelpMessage = "Projects to deploy", Mandatory = $false)]
-    [string] $projects = '',
     [Parameter(HelpMessage = "Name of environment to deploy to", Mandatory = $true)]
     [string] $environmentName,
     [Parameter(HelpMessage = "Artifacts to deploy", Mandatory = $true)]
     [string] $artifacts,
     [Parameter(HelpMessage = "Type of deployment (CD or Publish)", Mandatory = $false)]
     [ValidateSet('CD','Publish')]
-    [string] $type = "CD"
+    [string] $type = "CD",
+    [Parameter(HelpMessage = "The settings for all Deployment Environments", Mandatory = $true)]
+    [string] $deploymentEnvironmentsJson
 )
 
 $telemetryScope = $null
 
-if ($projects -eq '') {
-    Write-Host "No projects to deploy"
-    exit
-}
-
 try {
     . (Join-Path -Path $PSScriptRoot -ChildPath "..\AL-Go-Helper.ps1" -Resolve)
     DownloadAndImportBcContainerHelper
 
     import-module (Join-Path -path $PSScriptRoot -ChildPath "..\TelemetryHelper.psm1" -Resolve)
     $telemetryScope = CreateScope -eventId 'DO0075' -parentTelemetryScopeJson $parentTelemetryScopeJson
 
-    $EnvironmentName = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($environmentName))
+    $deploymentEnvironments = $deploymentEnvironmentsJson | ConvertFrom-Json | ConvertTo-HashTable -recurse
+    $deploymentSettings = $deploymentEnvironments."$environmentName"
+    $envName = $environmentName.Split(' ')[0]
+    $secrets = $env:Secrets | ConvertFrom-Json
+
+    # Check obsolete secrets
+    "$($envName)-EnvironmentName","$($envName)_EnvironmentName","EnvironmentName" | ForEach-Object {
+        if ($secrets."$_") {
+            throw "The secret $_ is obsolete and should be replaced by using the EnvironmentName property in the DeployTo$envName setting in .github/AL-Go-Settings.json instead"
```

Comment:
Isn't it better to check for obsolete settings/secrets as soon as possible? E.g. in `ReadSettings` and `ReadSecrets`.

Reply:
Environment secrets are not avaible before we are in the deploy step with the environment defined.
I check the obsolete settings early - but the secrets - no can do.